src/Symfony/Component/Security/Core/User/ChainUserChecker.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\User;
+
+/**
+ * Supports multiple user checkers.
+ *
+ * This user checker is a collection of other user checkers
+ * and triggers each user checker in the sequence provided.
+ *
+ * @author Iltar van der Berg <ivanderberg@hostnet.nl>
+ */
+final class ChainUserChecker implements UserCheckerInterface
```

Comment:
well, it certainly can't hurt - someone could always ask us to *not* make it final in the future. Anyways, we'll see what others think.